File: 2020/01/01.py
import itertools

lines = []
with open("input.txt") as f:
    for l in f.readlines():
        lines.append(int(l))

def two_makes_2020(ls):
    for a, b in itertools.combinations(ls, 2):
        if a + b == 2020:
            return a, b

def three_makes_2020(ls):
    for a, b, c in itertools.combinations(ls, 3):
        if a + b + c == 2020:
            return a, b, c

a, b = two_makes_2020(lines)
print("Part 1:", a * b)

a, b, c = three_makes_2020(lines)
print("Part 2:", a * b * c)

File: 2020/01/01.go
package main

import "io/ioutil"
import "fmt"
import "strconv"
import "strings"

func one(lines []int) int {
	for a := 0; a < len(lines); a++ {
		for b := a + 1; b < len(lines); b++ {
			if lines[a]+lines[b] == 2020 {
				return lines[a] * lines[b]
			}
		}
	}
	return 0
}

func two(lines []int) int {
	for a := 0; a < len(lines); a++ {
		for b := a + 1; b < len(lines); b++ {
			for c := b + 1; c < len(lines); c++ {
				if lines[a]+lines[b]+lines[c] == 2020 {
					return lines[a] * lines[b] * lines[c]
				}
			}
		}
	}
	return 0
}

func main() {
	data, _ := ioutil.ReadFile("input.txt")

	lines := []int{}
	for _, s := range strings.Split(string(data), "\n") {
		if i, err := strconv.Atoi(s); err == nil {
			lines = append(lines, i)
		}
	}

	fmt.Println("Part 1:", one(lines))
	fmt.Println("Part 2:", two(lines))
}
